src/keyfile.c

```diff
@@ -628,34 +631,45 @@ static void save_ui_prefs(GKeyFile *config)
 	}
 }
 
+typedef enum ConfigPayload
+{
+	PREFERENCES,
+	SESSION
+}
+ConfigPayload;
 
-void configuration_save(void)
+void write_config_file(gchar const *filename, ConfigPayload payload)
 {
 	GKeyFile *config = g_key_file_new();
-	gchar *configfile = g_build_filename(app->configdir, "geany.conf", NULL);
+	gchar *configfile = g_build_filename(app->configdir, filename, NULL);
 	gchar *data;
 
 	g_key_file_load_from_file(config, configfile, G_KEY_FILE_NONE, NULL);
 
 	/* this signal can be used e.g. to prepare any settings before Stash code reads them below */
 	g_signal_emit_by_name(geany_object, "save-settings", config);
```

Comment:
The signal should be emitted only once, arguably for the PREFERENCES payload. Might add another signal for save-session?

Reply:
Probably ok with just the prefs, not aware of anything in Geany or any plugins that use the `save-settings` for getting the session or MRUs.  Since they would have to change to the new signal anyway, leave it for now and add if they ask.

Reply:
Added in e406c69f (also fixed the "double vision" document pane bug). Like with the "load-settings" case, I'll leave this unresolved for you to check over the diff.